support-frontend/assets/helpers/productPrice/priceDescriptions.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import { fixDecimals } from 'helpers/subscriptions';
+import type { BillingPeriod } from 'helpers/billingPeriods';
+import {
+  billingPeriodNoun,
+  Quarterly,
+  SixWeekly,
+} from 'helpers/billingPeriods';
+import type { ProductPrice } from 'helpers/productPrice/productPrices';
+import {
+  getAppliedPromo,
+  hasDiscount,
+} from 'helpers/productPrice/productPrices';
+
+const displayPrice = (glyph: string, price: number) => `${glyph}${fixDecimals(price)}`;
+
+const billingPeriodQuantifier = (numberOfBillingPeriods: number, noun: string) =>
+  (numberOfBillingPeriods > 1 ?
+    `/${noun} for ${numberOfBillingPeriods} ${noun}s` :
+    ` for 1 ${noun}`);
+
+const billingPeriodDescription = (billingPeriod: BillingPeriod) =>
+  billingPeriodNoun(billingPeriod).toLowerCase();
+
+const standardRate = (glyph: string, price: number, billingPeriod: BillingPeriod) => `${displayPrice(glyph, price)} every ${billingPeriodDescription(billingPeriod)}`;
+
+function getDiscountDescription(
+  glyph: string,
+  price: number,
+  discountedPrice: number,
+  numberOfDiscountedPeriods: ?number,
+  billingPeriod: BillingPeriod,
+) {
+  const noun = billingPeriodDescription(billingPeriod);
```

Comment:
'noun' seems a bit generic - is it possible to use a more descriptive term?

Reply:
I think noun is appropriate though because is the noun version of the billing period eg. year, month etc. I'll rename the function instead, that is really only a little helper to avoid having to call `toLowerCase` in multiple places.